src/openmc_fusion_benchmarks/benchmark.py

```diff
@@ -1,119 +1,180 @@
-        download_geometry(self.name, 'h5m', self.run_option, cwd)
-
-    def download_weight_windows(self, cwd: str = None):
-        # file needs to go on drive with the rest
-
-        # download ww file:
-        # return openmc.wwinp_to_wws(path/to/ww_file)
+    @abstractmethod
+    def build_materials(self):
+        """Build materials for the benchmark."""
         pass
 
-    def _run_and_store(self):
-        pass
-
-
-def _wrap_run(download_files_func, original_run):
-    """Standalone function to wrap `run()` and ensure files are downloaded first."""
-    @wraps(original_run)
-    def wrapped_run(*args, **kwargs):
-        cwd = kwargs.get("cwd", ".")  # Extract cwd argument (default: ".")
-        download_files_func(cwd)  # Ensure files are downloaded
-        return original_run(*args, **kwargs)  # Call the original method
-    return wrapped_run
+    # @abstractmethod
+    # def build_geometry(self):
+    #     """Build geometry for the benchmark."""
+    #     return self._benchmark_spec['geometry']
 
+    @abstractmethod
+    def build_source(self):
+        """Build settings for the benchmark."""
+        pass
 
-class FngStr(Benchmark):
-    def __init__(self, run_option: str = 'onaxis'):
-        super().__init__("fng_str")
-
-        self.run_option = run_option
-
-
-class FngW(Benchmark):
-    def __init__(self, run_option: str = 'reaction_rates'):
-        super().__init__("fng_w")
-
-        self.run_option = run_option
-
-
-class Oktavian(Benchmark):
-    def __init__(self, run_option: str = 'Al'):
-        super().__init__("oktavian")
-
-
-class FnsDuct(Benchmark):
-    def __init__(self):
-        super().__init__("fns_duct")
-
+    # @abstractmethod
+    # def build_settings(self):
+    #     """Build settings for the benchmark."""
+    #     pass
 
-class FnsCleanW(Benchmark):
-    def __init__(self):
-        super().__init__("fns_clean_w")
+    # @abstractmethod
+    # def build_tallies(self):
+    #     """Build tallies for the benchmark."""
+    #     pass
 
 
-class BenchmarkDatabase:
-    @staticmethod
-    def get_benchmark(name: str, **kwargs):
-        if name == "fng_str":
-            return FngStr(**kwargs)
-        elif name == "fng_w":
-            return FngW(**kwargs)
-        elif name == "oktavian":
-            return Oktavian(**kwargs)
-        elif name == "fns_duct":
-            return FnsDuct(**kwargs)
-        elif name == "fns_clean_w":
-            return FnsCleanW(**kwargs)
-        else:
-            return Benchmark(name)
+class OpenmcBenchmark(Benchmark):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self._materials = None
+        self._geometry = None
+        self._settings = None
+        self._tallies = None
+
+    def build_materials(self):
+        # Implement the logic to build materials for OpenMC
+        material_data = self._benchmark_spec['materials']
+
+        fraction_map = {'atomic': 'ao', 'weight': 'wo'}
+
+        materials = []
```

Comment:
```suggestion
        materials = openmc.Materials()
```
The `Materials` class behaves like a list so we can just use it directly here.

Reply:
done